docs/kap_proposals/kap_7_remote_inventory.md

```diff
@@ -0,0 +1,64 @@
+# Remote Inventory Federation
+
+This feature would add the ability to Kapitan to fetch parts of the inventory from remote locations (https/git). This would allow users to combine different inventories from different sources and build modular infrastructure reusable across various repos.
+
+
+Author: @alpharoy14
+
+## Specification
+
+The configuration and declaration of remote inventories would be done in the hidden file `​.kapitan`.
+
+The file specifications are as follows:
+
+```yaml
```

Comment:
How about using this spec instead in the inventory itself? This could go under the `inventory` key (since nothing else uses it, to me is makes it very clear that its remote. It would avoid duplicating similar functionality we already implemented on top of the inventory and you could keep the semantics you've suggested later on in your proposal.

Reply:
I believe this will become something similar to external dependency management. The target files would have an option of `inventory` key. Something like 
``` yaml
parameters:
  kapitan:
    inventory:
    - name: <inventory_name>
      type: <inventory_type>
      output_path: relative/path/to/file/or/dir
      source: <source_of_inventory>
```
The differentiating factor from remote dependency would be using ` $ kapitan inventory --fetch <inventory-name> `  before compiling. This way the user can edit the fetched inventory configurations. Did I understand it correctly?

Reply:
@ramaro how will that work if inventory fetching is done in a separate step before running reclass?
We'd have to build the inventory twice. Once before fetching external deps and once after fetching them? 
Or am I missing something?

Reply:
We will only need to run it twice if there's something new to fetch - which we can detect, right?

Reply:
And yes, the other option is to make fetching manual using `--fetch`. But that wouldn't be consistent with the behaviour of `kapitan compile --fetch`, so I guess it makes sense to have `--fetch` for force fetch only if we implement a way of detecting that an remote inventory hasn't been fetched at all (e.g. check the output_path doesn't exist)

Reply:
Just a thought: manual fetching might turn out to be useful as users might want to tweak certain parameters before compiling. Besides this, if the entire process is automated then we might face the problem of infinite loop if the inventory is force compiled (using the `--fetch` flag).

Reply:
I think that could be easy to detect if it happens! Willing to take the risk for the sake of simplicity now. We can revisit if it is really an issue.

Reply:
great! So the way I understand it, it's similar to how external dependency is implemented except for the case of remote inventory, the `inventory` key will be parsed first to download and save the inventories in the `output_path` and then the `dependency` key will be parsed to download the remote dependencies. Did I miss something? If not, should I update this in the proposal and start with the implementation?

Reply:
That sgtm! :)